Physics/Assets/Scripts/Camera/CustomCamera.cs

```diff
@@ -0,0 +1,143 @@
+    }
+
+    private void takeScreenshot(string folder, Vector2Int screenshotSize)
+    {
+        cam.enabled = false;
+        var renderTexture = new RenderTexture(screenshotSize.x, screenshotSize.y, 24);
+        cam.targetTexture = renderTexture;
+
+        // Render the camera's view.
+        cam.Render();
+        RenderTexture.active = renderTexture;
+
+        // Make a new texture and read the active Render Texture into it.
+        Texture2D image = new Texture2D(screenshotSize.x, screenshotSize.y, TextureFormat.RGB24, false);
+        image.ReadPixels(new Rect(0, 0, screenshotSize.x, screenshotSize.y), 0, 0);
+        image.Apply();
+
+        Destroy(renderTexture);
+
+        // Replace the original active Render Texture.
+        cam.targetTexture = null;
+        RenderTexture.active = null;
+        cam.enabled = true;
+
+        // now image holds the image in texture2d form
+        byte[] png = ImageConversion.EncodeToPNG(image);
+
+        var filename = generateScreenshotPath(folder);
+
+        File.Open(filename, FileMode.OpenOrCreate).Close();
+        File.WriteAllBytes(filename, png);
+        Debug.LogFormat("Screenshot saved to {0}.", filename);
+    }
+
+    IEnumerator screenshotCoroutine(string folder, Vector2Int screenshotSize, float delay)
+    {
+        while (true)
+        {
+            takeScreenshot(folder, screenshotSize);
+            yield return new WaitForSecondsRealtime(delay);
+        }
+    }
+
+    public void SaveCamera()
+    {
+        cam = this.GetComponent<Camera>();
+        if (cam == null) Debug.LogError("broke");
+    }
+
+    private void OnEnable()
+    {
+        SaveCamera();
+        if (cam == null)
+        {
+            this.GetComponent<CustomCamera>().enabled = false;
+        }
+    }
+
+    public static void Screenshot(string folder, Vector2Int screenshotSize = default(Vector2Int))
+    {
+        if (screenshotSize.x < 300 || screenshotSize.y < 300)
+        {
+            // Set default size
+            screenshotSize = new Vector2Int(Screen.width, Screen.height);
+        }
+
+        cam.gameObject.GetComponent<CustomCamera>().takeScreenshot(folder, screenshotSize);
+    }
+
+    public void TakeContinuousScreenshots(string folder, 
+        Vector2Int screenshotSize = default(Vector2Int), float delay = 2f)
+    {
+        _screenshotCoroutine = 
+            StartCoroutine(screenshotCoroutine(folder, screenshotSize, delay));
+    }
+
+    public void StopContinuousScreenshots() {
+        if (_screenshotCoroutine != null)
+        {
+            StopCoroutine(_screenshotCoroutine);
+        } else
+        {
+            Debug.Log("No Screenshot Coroutine is active.");
+        }   
+    }
+
+    [RuntimeInitializeOnLoadMethod]
+    public static void AttachToCamera()
+    {
+        Scene currentScene = SceneManager.GetActiveScene();
+        List<GameObject> rootObjects = new List<GameObject>();
+        currentScene.GetRootGameObjects(rootObjects);
```

Comment:
Consider using FincObjectOfType<T>() to replace/simplify manual implementation here.

https://docs.unity3d.com/2020.1/Documentation/ScriptReference/Object.FindObjectOfType.html

Reply:
Replaced with `FindObjectOfType<Camera>()`;